src/main/kotlin/apply/domain/judgment/Judgment.kt

```diff
@@ -0,0 +1,82 @@
+package apply.domain.judgment
+
+import support.domain.BaseRootEntity
+import java.time.LocalDateTime
+import javax.persistence.CascadeType
+import javax.persistence.Column
+import javax.persistence.Entity
+import javax.persistence.EnumType
+import javax.persistence.Enumerated
+import javax.persistence.ForeignKey
+import javax.persistence.JoinColumn
+import javax.persistence.OneToMany
+
+@Entity
+class Judgment(
+    @Column(nullable = false)
+    val assignmentId: Long,
+
+    @Column(nullable = false)
+    @Enumerated(EnumType.STRING)
+    val type: JudgmentType,
+    records: List<JudgmentRecord> = emptyList(),
+    id: Long = 0L
+) : BaseRootEntity<Judgment>(id) {
+    @OneToMany(cascade = [CascadeType.PERSIST, CascadeType.REMOVE])
+    @JoinColumn(
+        name = "judgment_id", nullable = false, updatable = false,
+        foreignKey = ForeignKey(name = "fk_judgment_record_judgment_id_ref_judgment_id")
+    )
+    private val records: MutableList<JudgmentRecord> = records.toMutableList()
+
+    val lastCommit: Commit
+        get() = lastRecord.commit
+
+    val lastStatus: JudgmentStatus
+        get() = lastRecord.status
+
+    val lastRecord: JudgmentRecord
+        get() = records.maxByOrNull { it.startedDateTime } ?: throw NoSuchElementException()
+
+    fun start(commit: Commit) {
+        check(canStart()) { "자동 채점을 시작할 수 없습니다." }
+        val record = findRecord(commit) ?: createRecord(commit)
+        if (record.completed) {
+            record.touch()
+        } else {
+            record.start()
+            registerEvent(JudgmentStartedEvent(id, commit))
+        }
+    }
+
+    private fun canStart(): Boolean {
+        if (records.isEmpty()) return true
+        return lastRecord.startedDateTime.plusMinutes(DELAY_MINUTES) < LocalDateTime.now()
+    }
```

Comment:
```suggestion
    private fun canStart(): Boolean {
        return if (records.isEmpty()) true
        else lastRecord.startedDateTime.plusMinutes(DELAY_MINUTES) < LocalDateTime.now()
    }
```
a: return을 묶어 쓰는건 어떤가요?

Reply:
추후에 어떤 조건이 추가될지는 모르겠지만 early return이 더 나을 것 같아요. 참고로 `else`가 같은 줄에 없으면 중괄호가 필요합니다. 아래의 Kotlin style guide를 참고해 주세요.
https://developer.android.com/kotlin/style-guide?hl=ko#braces